qa/os/src/test/java/org/elasticsearch/packaging/test/KeystoreManagementTests.java

```diff
@@ -260,6 +262,7 @@ public void test51WrongKeystorePasswordFromFile() throws Exception {
      * Check that we can mount a password-protected keystore to a docker image
      * and provide a password via an environment variable.
      */
+    @AwaitsFix(bugUrl = "Keystore fails to save with resource busy")
```

Comment:
@williamrandolph apart from the general feedback on the packaging tests changes, if you have any idea why these fail in such a way if I pass the ELASTIC_PASSWORD env var, I'd be more than grateful

Reply:
I don't know off the top of my head, but I'll take a look!

Reply:
I figured it out.

It's because the keystore is bind mounted (to be password protected) and adding values to the keystore (to write the bootstrap.password) replaces the bind mounted keystore file. This is not permitted (Device or resource busy on keystore save).
I've found resources online saying that writing to bind mounted files is OK but I've not found much details about replacing, so I've started the container (which has a bind mounted keystore) without specifying the bootstrap.password env variable and then tried to add values to the keystore. It fails the same way.

Reply:
I think the fix is to bind mount the full config dir, but I haven't tested it.

Reply:
Thanks mate, I tried this out in 364a984, let's see what CI has to say

Reply:
Looks like this was it and your suggested workaround works :rocket:

Reply:
Ooops, looks like you've forgot to unmute the tests...

Turns out there is a bug in the cmd line tools. They create multiple input stream readers from the same input stream. This makes the first reader consume more output than it needs.

I've tracked it down, and now all these tests are passing. Here's the diff: https://github.com/jkakavas/elasticsearch/compare/security-enabled-default...albertzaharovits:fix_keystore_tests?expand=1

Feel free to update the tests here, and I will open a separate PR for the cmd line tool.

Reply:
> Ooops, looks like you've forgot to unmute the tests... 

Talk about wishful thinking... I added your changes and will wait for your PR before we unmute the tests

Reply:
Raised https://github.com/elastic/elasticsearch/pull/76124 .